modules/social_features/social_event/social_event.module

```diff
@@ -638,3 +638,48 @@ function social_event_get_current_event() {
 
   return NULL;
 }
+
+/**
+ * Implements hook_preprocess_field().
+ */
+function social_event_preprocess_field(&$variables) {
+  $formatter = $variables['element']['#formatter'];
+  if (in_array($formatter, ['address_plain', 'address_default'])) {
+    $entity = $variables['element']['#object'];
+    if ($entity && $entity instanceof NodeInterface && $entity->getType() === 'event') {
+      $social_event_settings = \Drupal::config('social_event.settings');
+      $address_visibility_settings = $social_event_settings->get('address_visibility_settings');
+      if (isset($address_visibility_settings['street_code_private']) && !empty($address_visibility_settings['street_code_private'])) {
+        $enrollments = \Drupal::entityTypeManager()->getStorage('event_enrollment')->loadByProperties([
+          'field_account' => \Drupal::currentUser()->id(),
+          'field_event' => $entity->id(),
+        ]);
+
+        $hide_street_and_postal_code = FALSE;
+        if ($enrollment = array_pop($enrollments)) {
+          $enrollment_status = $enrollment->field_enrollment_status->value;
+          if ($enrollment_status === '0') {
+            $hide_street_and_postal_code = TRUE;
+          }
+        }
+        else {
+          $hide_street_and_postal_code = TRUE;
+        }
+
+        if ($hide_street_and_postal_code) {
+          switch ($formatter) {
+            case 'address_plain':
+              $variables['items'][0]['content']['#address_line1'] = NULL;
```

Comment:
@denysyukm Maybe it would be good to add some check if this variable exists? Just in case if something will be changed in future.

Reply:
Added